### ISSUE TITLE
Proposal: Encoding a temporal extent

### ISSUE THREAD

**Issue by cportele** [opengeospatial/ogc-feat-geo-json#11]
Many features have a spatial geometry that provides information about the location of the feature. In GeoJSON, this information is encoded in the top-level "geometry" member. Features are often associated with temporal information, too. In most cases this is either an instant (e.g., an event) or an interval (e.g., an activity or a temporal validity). In OGC API Features this is reflected in the `datetime` parameter for temporal filtering of features.

The proposal adds support for the most common case: associating a feature with a single temporal instant or interval in the Gregorian calendar.

More complex cases and other temporal coordinate reference systems are out-of-scope of this proposal and might be specified in extensions.

https://github.com/opengeospatial/ogc-feat-geo-json/blob/main/proposals/temporal-extent.adoc

**Comment by ilkkarinne** [comment]
The O&M SWG would like to hear why the temporal extent is proposed to be encoded using a non-standard JSON object for instants and intervals rather than the standard and widely used ISO 8601-1:2019 string formats.

This question was raised in the OMS JSON workshop on 29th Sep 2021. The workshop participants preferred using the ISO 8601-1:2019 and ISO 8601-2:2019 formats below for the OMS v3.0 JSON implementation temporal properties, and would be happier to see also the FG JSON ```when``` property to use the existing standards rather than requiring supporting two different temporal property encodings:

- instant: ```2021-10-01T08:54:31+00:00```
- interval with start and end times: ```2007-03-01T13:00:00Z/2008-05-11T15:30:00Z```
- interval with start time and duration: ```2007-03-01T13:00:00Z/P1Y2M10DT2H30M```
- interval with duration and end time: ```P1Y2M10DT2H30M/2008-05-11T15:30:00Z```
- interval with an open ending : ```2007-03-01T13:00:00Z/..``` (ISO 8601-2)
- interval with an open beginning : ```../2007-03-01T13:00:00Z``` (ISO 8601-2)

**Comment by cportele** [comment]
The main reason for the current design is that this is easier to parse. ISO 8601 is useful whenever a text representation is necessary or best (like in the `datetime` OGC API query parameter), but in a structured language like JSON it is natural to be explicit and use the structures. This is why GeoJSON does not represent the geometry as a WKT string, but as a JSON structure. The JSON encoding of the candidate Common Query Language standard follows that approach, too, for spatial and temporal geometries.

Another reason is that the current design would also allow to support other temporal types in the future besides instants and intervals. For example, a time series with an array of timestamps.

One question: You mention "a non-standard JSON object for instants and intervals". Are you aware of any standard JSON object that we could reuse?

**Comment by hylkevds** [comment]
> ISO 8601 is useful whenever a text representation is necessary or best (like in the `datetime` OGC API query parameter), but in a structured language like JSON it is natural to be explicit and use the structures.

Interesting argument, especially since you are still using ISO 8601 for all other aspects of time, only the interval/instant encoding of ISO 8601 is explicitly left out. So the user will still have to parse ISO 8601 time strings.

**Comment by cportele** [comment]
> So the user will still have to parse ISO 8601 time strings.

Not really. I think all environments have standard functions that do the parsing / generating of such timestamps for you, in particular since this is not about all the ISO 8601 options, but the restricted RFC 3339 profile for the internet. 

In CQL2 we restrict all timestamp literals to UTC to make processing as straightforward as possible and maybe we should do this here, too.

**Comment by KathiSchleidt** [comment]
Limiting to UTC will cause issues with use communities, would prefer to clearly require time-zone. From experience I also worry about the use of "Z" to indicate UTC, while we understand this, users don't, and as they don't see a time-zone, they end up providing their local time with a Z (if there's a number there, they have to start thinking!)

**Comment by cportele** [comment]
Meeting 2021-10-04: 

* We discussed the comment, but confirmed that the SWG thinks that a structured approach for anything beyond RFC 3339 is the right approach.
* Regarding the time zone, we confirmed that every timestamp must have a timezone. Whether the timezone has to be UTC, at least in a Part 1, can be up for discussion. We note that GeoPackage, CDB and CQL2 restrict literals to UTC with "Z" and a consistent approach that reduces the degrees of freedom is helpful for developers. There was also an OAB discussion, Carl will look for the notes and provide feedback.
* Maybe we also need to clarify that the instant/interval objects are only required for the "when" member, but we do not prescribe how temporal attributes are encoded in the properties object. For example, an attribute that is always a date can still be encoded as a RFC 3339 date string.
